### ISSUE TITLE
Provide option to install node using Brew

### ISSUE THREAD

**Issue by amosshapira** [voxpupuli/puppet-nodejs#236]
When using the module to install Node on OSX, I get the following errors:

```
Error: /Package[nodejs]: Provider macports is not functional on this host
Error: /Package[nodejs-devel]: Provider macports is not functional on this host
Error: /Package[npm]: Provider macports is not functional on this host
```

I use Homebrew to install node on the system. I don't use Macports anywhere on our network.
I need a way to tell the module to use Brew instead of Macports.

**Comment by kizzie** [comment]
So this seems to be annoyingly easy and really hard at the same time. To get it to use homebrew you can edit the params.pp file and change line 171 to read

``` Package { provider => 'homebrew' } ```

Problem is I can't think of an elegent way to let people set hat value as the params class doesn't take inputs and moving the entire case statement out of the params.pp seems a little extreme for a simple fix pull request.

**Comment by kizzie** [comment]
Have a vague idea at a PR - https://github.com/voxpupuli/puppet-nodejs/pull/321

Cause either I need to do one to update the macports package name (its now nodejs6|7|8 or allow for homebrew...

**Comment by juniorsysadmin** [comment]
Fixed by #321
